### PR TITLE
feat: Adding WindowsNodeReset.ps1 script to reset/cleanup state for nodes

### PR DESCRIPTION
**Pull request by marosset**
**Reason for Change**:

**Issue Fixed**:

**Requirements**:

- [x] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Notes**:

### CODE REVIEW

File: parts/k8s/kuberneteswindowsfunctions.ps1 [hunk 1]
```diff
@@ -137,3 +137,19 @@ function Get-NetworkLogCollectionScripts {
     DownloadFileOverHttp -Url 'https://github.com/microsoft/SDN/raw/master/Kubernetes/windows/debug/stoppacketcapture.cmd' -DestinationPath 'c:\k\debug\stoppacketcapture.cmd'
     DownloadFileOverHttp -Url 'https://github.com/microsoft/SDN/raw/master/Kubernetes/windows/helper.psm1' -DestinationPath 'c:\k\debug\helper.psm1'
 }
+
+function Register-NodeResetScriptTask {
+    Write-Log "Creatingg a startup task to run windowsnodereset.ps1"
```
Comment (MikkelHegn) [inline parts/k8s/kuberneteswindowsfunctions.ps1]: Minor spelling nit _"Creatingg"_

**Comment by marosset** [comment]
cc @ksubrmnn

File: parts/k8s/windowsnodereset.ps1 [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+Write-Log "Stopping kubelet service"
+Stop-Service kubelet
+
+#
+# Perform cleanup
+#
+
+$hnsNetwork = Get-HnsNetwork | Where-Object Name -EQ azure
+if ($hnsNetwork) {
+    Write-Log "Cleaning up HNS network 'azure'..."
```
Comment (ksubrmnn) [inline parts/k8s/windowsnodereset.ps1]: nit: This log makes me think you're deleting the HNS network here but you do it after container cleanup.

File: parts/k8s/windowsnodereset.ps1 [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+    foreach ($file in $filesToRemove) {
+        if (Test-Path $file) {
+            Write-Log "Deleting stale file at $file"
+            Remove-Item $file
+        }
+    }
+}
+
+#
+# Create required networks
```
Comment (ksubrmnn) [inline parts/k8s/windowsnodereset.ps1]: Would prefer more detailed comments. Not quite sure what the below section is supposed to be doing (from a newcomer's perspective)

File: parts/k8s/windowsnodereset.ps1 (hunk 2 above)
Comment (marosset) [reply parts/k8s/windowsnodereset.ps1]: We do delete the HNS network (line 46) we just delete the containers first.
I can update comments tho

File: parts/k8s/windowsnodereset.ps1 (hunk 3 above)
Comment (marosset) [reply parts/k8s/windowsnodereset.ps1]: Is the new context ok?

**Pull request merged**